spec/addressable/uri_spec.rb

```diff
@@ -6338,6 +6338,35 @@ def to_str
   end
 end
 
+describe Addressable::URI, "when given the input: 'user@domain.com'" do
+  before do
+    @input = "user@domain.com"
+  end
+
+  context "for heuristic parse" do
+    it "should remains 'mailto:user@domain.com'" do
+      uri = Addressable::URI.heuristic_parse("mailto:#{@input}")
+      expect(uri.to_s).to eq("mailto:user@domain.com")
+    end
+
+    it "should have a scheme of 'mailto'" do
+      uri = Addressable::URI.heuristic_parse(@input)
+      expect(uri.to_s).to   eq("mailto:user@domain.com")
+      expect(uri.scheme).to eq("mailto")
+    end
+
+    it "should remains 'acct:user@domain.com'" do
+      uri = Addressable::URI.heuristic_parse("acct:#{@input}")
+      expect(uri.to_s).to eq("acct:user@domain.com")
+    end
+
+    it "should ignore '@' if some scheme is present" do
+      uri = Addressable::URI.heuristic_parse("http://#{@input}/")
+      expect(uri.to_s).to eq("http://user@domain.com/")
```

Comment:
Add: `expect(uri.user).to eq("user")`

Reply:
I did two separate tests for the context of HTTP, I thought it would be better. Thank you